runtimes/common/src/dip/merkle/tests/generate_commitment.rs

```diff
@@ -0,0 +1,42 @@
+// KILT Blockchain – https://botlabs.org
+// Copyright (C) 2019-2024 BOTLabs GmbH
+
+// The KILT Blockchain is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// The KILT Blockchain is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+// If you feel like getting in touch with us, you can do so at info@botlabs.org
+
+use did::did_details::DidVerificationKey;
+use frame_support::assert_err;
+use pallet_dip_provider::traits::IdentityCommitmentGenerator;
+
+use crate::{
+	constants::dip_provider::MAX_LINKED_ACCOUNTS,
+	dip::{
+		merkle::{DidMerkleProofError, DidMerkleRootGenerator},
+		mock::{create_linked_info, TestRuntime, ACCOUNT, DID_IDENTIFIER},
+	},
+};
+
+#[test]
```

Comment:
The thing is that that struct exposes twi different functions. I liked the approach we have for pallets where each function gets its own file, especially if the test logic is complex or the test cases are a lot, as in this case. Otherwise, both functions being tested would end up in the same file, and it would easily get messy, I feel.

Reply:
In general, I agree. Just for this case, the tests are very small. The overhead of having them in separated files is from MPOV a bit overkill.

Reply:
Just to clarify, what do you think is the overhead, exactly?

Reply:
We introduce a mod file which consists of 10+ lines + the file itself which also consists of 10+ lines of code just for the imports, for a test which consists of 12 lines. And we have two test files. But the strict separation of tests and logic starts to grow on me. 

I just saw that a lot of times in other projects and libraries.